pallets/moonbeam-orbiters/src/lib.rs

```diff
@@ -0,0 +1,544 @@
+		_,
+		Twox64Concat,
+		T::RoundIndex,
+		Blake2_128Concat,
+		T::AccountId,
+		T::AccountId,
+		OptionQuery,
+	>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub min_orbiter_deposit: BalanceOf<T>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self {
+				min_orbiter_deposit: One::one(),
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			assert!(
+				self.min_orbiter_deposit > Zero::zero(),
+				"Minumal orbiter deposit should be greater tham zero"
+			);
+			MinOrbiterDeposit::<T>::put(self.min_orbiter_deposit)
+		}
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_: T::BlockNumber) -> Weight {
+			// Prune old OrbiterPerRound entries
+			let current_round = CurrentRound::<T>::get();
+			let max_round_archive = T::MaxRoundArchive::get();
+			if current_round > max_round_archive {
+				let round_to_prune = current_round - max_round_archive;
+				OrbiterPerRound::<T>::remove_prefix(round_to_prune, None);
+				T::DbWeight::get().reads_writes(1, 1)
+			} else {
+				T::DbWeight::get().reads(1)
+			}
+		}
+	}
+
+	/// An error that can occur while executing this pallet's extrinsics.
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The collator is already added in orbiters program.
+		CollatorAlreadyAdded,
+		/// This collator is not in orbiters program.
+		CollatorNotFound,
+		/// There are already too many orbiters associated with this collator.
+		CollatorPoolTooLarge,
+		/// There are more collator pools than the number specified in the parameter.
+		CollatorsPoolCountTooLow,
+		/// The minimum deposit required to register as an orbiter has not yet been included in the
+		/// onchain storage
+		MinOrbiterDepositNotSet,
+		/// This orbiter is already associated with this collator.
+		OrbiterAlreadyInPool,
+		/// This orbiter has not made a deposit
+		OrbiterDepositNotFound,
+		/// This orbiter is not found
+		OrbiterNotFound,
+		/// The orbiter is still at least in one pool
+		OrbiterStillInAPool,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// An orbiter join a collator pool
+		OrbiterJoinCollatorPool {
+			collator: T::AccountId,
+			orbiter: T::AccountId,
+		},
+		/// An orbiter leave a collator pool
+		OrbiterLeaveCollatorPool {
+			collator: T::AccountId,
+			orbiter: T::AccountId,
+		},
+		/// Paid the orbiter account the balance as liquid rewards.
+		OrbiterRewarded {
+			account: T::AccountId,
+			rewards: BalanceOf<T>,
+		},
+		OrbiterRotation {
+			collator: T::AccountId,
+			old_orbiter: Option<T::AccountId>,
+			new_orbiter: Option<T::AccountId>,
+		},
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Add an orbiter in a collator pool
+		#[pallet::weight(T::WeightInfo::collator_add_orbiter())]
+		pub fn collator_add_orbiter(
```

Comment:
I think that a hint is not worth it here, it would complicate the usage for a negligible gain.
Because the number of reads/writes does not vary, and in practice the max size of a pool will be small (8).